sources/packages/backend/apps/queue-consumers/src/processors/schedulers/base-queue.ts

```diff
@@ -0,0 +1,74 @@
+import { CustomNamedError, parseJSONError } from "@sims/utilities";
+import {
+  InjectLogger,
+  LoggerService,
+  ProcessSummary,
+} from "@sims/utilities/logger";
+import {
+  PROCESS_SUMMARY_CONTAINS_ERROR,
+  logProcessSummaryToJobLogger,
+} from "../../utilities";
+import { Job } from "bull";
+import { Process } from "@nestjs/bull";
+
+/**
+ * Provides basic functionality for queue processing.
+ */
+export abstract class BaseQueue<T> {
```

Comment:
It's good to have it in a abstract class. We can even have the logging for the job initialization and job ending in abstract methods if we that level of customization for some process. Good idea!